spec/features/cases/offender_sar_complaint/case_creating_spec.rb

```diff
@@ -286,55 +348,38 @@ def and_confirm_offender_sar_case
     click_on "Continue"
   end
 
-  def and_fill_in_complaint_type_page(params = nil)
+  def and_fill_in_complaint_type_page
     expect(cases_new_offender_sar_complaint_complaint_type_page).to be_displayed
-    actual_params = generate_options_on_complaint_type_page(params)
-    cases_new_offender_sar_complaint_complaint_type_page.fill_in_case_details(actual_params)
+    cases_new_offender_sar_complaint_complaint_type_page.fill_in_case_details(default_params)
     click_on "Continue"
   end
 
-  def generate_options_on_complaint_type_page(params)
-    # In order to get good coverage for the combination of complaint_type, complaint_subtype, priority
-    # without going through the whole sample space, a simple approatch is to pick up the choice randomly
-    actual_params = params || {}
-    @chosen_complaint_type = actual_params[:complaint_type] || 
-                Case::SAR::OffenderComplaint.complaint_types.keys().sample(1)[0]
-    @chosen_complaint_subtype = actual_params[:complaint_subtype] || 
-                Case::SAR::OffenderComplaint.complaint_subtypes.keys().sample(1)[0]
-    @chosen_complaint_priority = actual_params[:priority] || 
-                Case::SAR::OffenderComplaint.priorities.keys().sample(1)[0]
-    actual_params[:complaint_type] = @chosen_complaint_type
-    actual_params[:complaint_subtype] = @chosen_complaint_subtype
-    actual_params[:priority] = @chosen_complaint_priority
-    actual_params
-  end
-
```

Comment:
as I mentioned in my previous comments,  I would prefer to generate the choices of types within the test spec, maybe I misunderstood your intention here,  discuss further?

Reply:
Yeah, sorry! I should have talked to you about this before PR. See other reply though :)  

Another wrinkle is I wanted to use let rather than instance variables and check the expectations based on an actual case object rather than fine-grained values; the main problem is this code would need to set contact details as well as setting the complaint type, subtype and priority. On the whole, I think we should use factories and traits to cover variations and have specific named scenarios that we care about, rather than dynamically generating the coverage